generator/src/test/java/com/scottlogic/deg/generator/cucumber/MatchingRegex.feature

```diff
@@ -577,5 +577,5 @@ Scenario: Running a 'matchingRegex' request as part of an if constraint should b
       | "d"  | 2     |
       | null | 1     |
       | null | 2     |
-      | null | null  |
-      | null | null  |
+      | "a"  | null  |
```

Comment:
Hi @pdaulbyscottlogic why has the expected values of this test changed? Based on the changes I was only expecting results to be reordered not changed. I must be missing something. Thanks!

Reply:
The reason for this test to have required changes is due to the order of the values emitted having changed and this being important in the PINNING combination strategy. See documentation: https://github.com/ScottLogic/data-engineering-generator/blob/master/generator/docs/CombinationStrategies.md#pinning

BEFORE the change, we were getting the following values as input to pinning:

foo: null,a,b (if)
price:1,null

foo: null,c,d (else)
price: 2,null


When we apply pinning combination strategy we get:

From if:
null,1
a,1
b,1
null,null

From else:
null,2
c,2
d,2
null,null


AFTER the change, we have demoted null so from if conditions it now appears after the letters e.g. c,d,null (rather than null,c,d)

foo: a,b,null (if)
price:1,null

foo: c,d,null (else)
price: 2,null


When we apply pinning combination strategy we get:

From if:
a,1
b,1
null,1
a,null

From else:
c,2
d,2
null,2
c,null

Reply:
The reason why we were getting 1,null and 2,null in both cases (null was demoted BEFORE and AFTER) is that on master, we were already demoting null for sets only.

Reply:
I am happy that we understand the reasoning behind this change now. @sl-slaing @pdaulbyscottlogic

Reply:
Ok, so my understanding is that `null` is emitted consistently as the last value in results now, whether there is a set restriction or not.